Library/Homebrew/formula.rb

```diff
@@ -2096,7 +2096,7 @@ def to_hash
       "version_scheme"           => version_scheme,
       "bottle"                   => {},
       "keg_only"                 => keg_only?,
-      "keg_only_reason"          => keg_only_reason&.to_hash,
+      "keg_only_reason"          => keg_only_reason.gsub(HOMEBREW_PREFIX, HOMEBREW_PREFIX_PLACEHOLDER)&.to_hash,
```

Comment:
I don't think this will work, actually, since `keg_only_reason` is a `KegOnlyReason` object and not a string. Something like this, maybe?

```suggestion
      "keg_only_reason"          => keg_only_reason.to_hash.to_h do |key, value|
        [key, value.gsub(HOMEBREW_PREFIX, HOMEBREW_PREFIX_PLACEHOLDER)]
      end
```

(my suggestion might not satisfy `brew style` so maybe this needs to be handled separately below, but something like that)

Reply:
That's quite horrible, I'm inclined to just not and go with the RuboCop. 😅